src/components/app/TabBar.js

```diff
@@ -51,6 +52,21 @@ class TabBar extends PureComponent {
   }
 }
 
+type Props = {
+  className?: string,
+  tabs: Tab[],
+  selectedTabName: string,
+  tabOrder: number[],
+  onSelectTab: string => mixed,
+  onChangeTabOrder: (number[]) => any,
+  +_mouseDownListener?: () => mixed,
```

Comment:
for what it's worth, I think `onChangeTabOrder` and `onSelectTab` should be returning mixed simply because in this component we don't care about the return value. That it's void or not makes no difference.

Especially this is sometimes useful for unit testing if the functions return a promise :)

Not a big deal though, because currently it's still theoretical and we can change it back later if we need it.